tests/ci/finish_check.py

```diff
@@ -80,20 +93,27 @@ def main():
                 continue
             else:
                 has_failure = True
+                error_cnt += 1
 
         ci_state = SUCCESS  # type: StatusType
+        description = "All checks finished"
         if has_failure:
             ci_state = FAILURE
+            description = f"All checks finished. {error_cnt} jobs failed"
+        elif has_workflow_failures:
+            ci_state = FAILURE
+            description = "All checks finished. Workflow has failures."
         elif has_pending:
             print("ERROR: CI must not have pending jobs by the time of finish check")
+            description = "ERROR: workflow has pending jobs"
             ci_state = FAILURE
 
         if ci_status.state == PENDING:
```

Comment:
we should put the reflection of it as 

```
if  ci_status.state != PENDING:
    return
```

Into line 71. The whole work is noop after that line in case of `ci_status.state != PENDING`

Reply:
I don't think we need this check at all

Reply:
if workflow is restarted it's fine to overwrite any other status.

I removed this if

Reply:
fair